src/core/views.py

```diff
@@ -2433,13 +2454,14 @@ def get_queryset(self, params_querydict=None):
                 for predicate in predicates:
                     query |= Q(predicate)
                 q_stack.append(query)
-        return self.order_queryset(
-            self.filter_queryset_if_journal(
-                self.queryset.filter(*q_stack)
-            )
-        ).exclude(
-            stage=submission_models.STAGE_UNSUBMITTED,
+        self.queryset = self.filter_queryset_if_journal(
+            self.queryset.filter(*q_stack)
         )
+        if hasattr(self.model, 'stage'):
+            self.queryset = self.queryset.exclude(
+                stage=submission_models.STAGE_UNSUBMITTED
+            )
```

Comment:
At a minimum we should be more explicit here and say what we mean:
`if self.model is submission_models.Article:`

But ideally,to achieve proper polymorphism, since this class is now meant to be used with different models, we need to turn it into an abstract class (`GenericFacetedListView`?) which implements this method in the lines of:

```python
class FacetedArticleListView(GenericFacetedListView):
    model = submission_models.Article
    def get_queryset(self):
        qs = super().get_queryset()
        return qs.filter(
            stage=submission_models.STAGE_UNSUBMITTED
        )

Reply:
Yes, you are right. I didn't change the inheritance structure because I wanted to make sure you and Andy were good with expanding the view past articles. I will go ahead and implement it properly.